core/slice.go

```diff
@@ -0,0 +1,484 @@
+		sl.domClient = MakeDomClient(domClientUrl)
+	}
+
+	sl.subClients = make([]*quaiclient.Client, 3)
+	// only set the subClients if the chain is not region
+	if types.QuaiNetworkContext != params.ZONE {
+		go func() {
+			sl.subClients = MakeSubClients(subClientUrls)
+		}()
+	}
+
+	return sl, nil
+}
+
+// HeaderChain retrieves the headerchain.
+func (sl *Slice) HeaderChain() *HeaderChain {
+	return sl.hc
+}
+
+// MakeDomClient creates the quaiclient for the given domurl
+func MakeDomClient(domurl string) *quaiclient.Client {
+	if domurl == "" {
+		log.Crit("dom client url is empty")
+	}
+	domClient, err := quaiclient.Dial(domurl)
+	if err != nil {
+		log.Crit("Error connecting to the dominant go-quai client", "err", err)
+	}
+	return domClient
+}
+
+// MakeSubClients creates the quaiclient for the given suburls
+func MakeSubClients(suburls []string) []*quaiclient.Client {
+	subClients := make([]*quaiclient.Client, 3)
+	for i, suburl := range suburls {
+		if suburl == "" {
+			log.Warn("sub client url is empty")
+		}
+		subClient, err := quaiclient.Dial(suburl)
+		if err != nil {
+			log.Crit("Error connecting to the subordinate go-quai client for index", "index", i, " err ", err)
+		}
+		subClients[i] = subClient
+	}
+	return subClients
+}
+
+// Config retrieves the slice's chain configuration.
+func (sl *Slice) Config() *params.ChainConfig { return sl.config }
+
+// Engine retrieves the header chain's consensus engine.
+func (sl *Slice) Engine() consensus.Engine { return sl.engine }
+
+// Append
+func (sl *Slice) Append(block *types.Block) error {
+
+	err := sl.engine.VerifyHeader(sl.hc, block.Header(), true)
+	if err != nil {
+		return err
+	}
+
+	order, err := sl.GetDifficultyOrder(block.Header())
+	if err != nil {
+		return err
+	}
+
+	_, err = sl.PCRC(block.Header(), order)
+	if err != nil {
+		return err
+	}
+
+	err = sl.hc.Append(block)
+	if err != nil {
+		return err
+	}
+
+	td, err := sl.CalcTd(block.Header())
+	if err != nil {
+		return err
+	}
+
+	// write the tds
+	rawdb.WriteTd(sl.hc.headerDb, block.Hash(), block.NumberU64(), td)
+
+	// We have a new possible head call HLCR to potentially set
+	reorg := sl.HLCR(sl.hc.GetTd(sl.hc.currentHeaderHash, sl.hc.CurrentHeader().Number64()), td)
+
+	if reorg {
+		err = sl.hc.SetCurrentHeader(block.Header())
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// The purpose of the Previous Coincident Reference Check (PCRC) is to establish
+// that we have linked untwisted chains prior to checking HLCR & applying external state transfers.
+// NOTE: note that it only guarantees linked & untwisted back to the prime terminus, assuming the
+// prime termini match. To check deeper than that, you need to iteratively apply PCRC to get that guarantee.
+func (sl *Slice) PCRC(header *types.Header, headerOrder int) (types.PCRCTermini, error) {
+
+	if header.Number[types.QuaiNetworkContext].Cmp(big.NewInt(0)) == 0 {
+		return types.PCRCTermini{}, nil
+	}
+
+	slice := header.Location
```

Comment:
This name collision is very confusing, now that this method acts on the `Slice` type. Maybe just call this variable `location`, since that is indeed what it is?